neuralpp/test/quick_tests/symoblic/expression_test.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+
+from neuralpp.symbolic.function import Add
+from neuralpp.symbolic.basic_expression import BasicFunctionApplication, BasicConstant, BasicVariable
+    
+
+def test_basic_constant():
+    # Constant can be anything
+    constant_one = BasicConstant(1)
+    constant_abc = BasicConstant("abc")
+    assert constant_one != constant_abc
+    assert constant_one == BasicConstant(2-1)
+    assert constant_one.subexpression() == []
+    assert not constant_one.contains(constant_abc)
+    assert not constant_one.contains(constant_one)  # A constant does not contains() itself
```

Comment:
I was thinking of the semantics of `contains` as similar to "subset". A set contains itself, so an expression should contain itself.

Reply:
got it.